terminus/geometry/arc.py

```diff
@@ -88,8 +109,34 @@ def find_intersection(self, other):
     def _find_arc_intersection(self, other):
         circle1 = Circle(self.center_point(), self.radius())
         circle2 = Circle(other.center_point(), other.radius())
-        candidates = circle1.intersection(circle2)
-        return filter(lambda point: self.includes_point(point) and other.includes_point(point), candidates)
+        if (circle1.center - circle2.center).norm() > 1e-5 or abs(circle1.radius - circle2.radius) > 1e-5:
```

Comment:
Aren't things like `(circle1.center - circle2.center).norm()` and `arc1.end_point().distance_to(arc2.start_point())` the same? If so, please use a single way of doing it (preferably `distance_to`).